src/Scaler/Services/CosmosDbMetricProvider.cs

```diff
@@ -3,47 +3,73 @@
+
+            using MeterProvider meterProvider = Sdk.CreateMeterProviderBuilder()
+               .AddMeter("OrderProcessor.CFStore")               
+               .Build();
         }
 
         public async Task<long> GetPartitionCountAsync(ScalerMetadata scalerMetadata)
         {
             try
             {
-                Container leaseContainer = _factory
-                    .GetCosmosClient(scalerMetadata.LeaseConnection)
-                    .GetContainer(scalerMetadata.LeaseDatabaseId, scalerMetadata.LeaseContainerId);
 
-                ChangeFeedEstimator estimator = _factory
-                    .GetCosmosClient(scalerMetadata.Connection)
-                    .GetContainer(scalerMetadata.DatabaseId, scalerMetadata.ContainerId)
-                    .GetChangeFeedEstimator(scalerMetadata.ProcessorName, leaseContainer);
+                var credential = new DefaultAzureCredential();
+                var cosmosClient = new CosmosClient(scalerMetadata.Endpoint, credential);
+
+
+                Container leaseContainer = cosmosClient.GetContainer(scalerMetadata.LeaseDatabaseId, scalerMetadata.LeaseContainerId);
+                Container monitoredContainer = cosmosClient.GetContainer(scalerMetadata.DatabaseId, scalerMetadata.ContainerId);
 
-                // It does not help by creating more change-feed processing instances than the number of partitions.
+
+                ChangeFeedEstimator changeFeedEstimator = monitoredContainer.GetChangeFeedEstimator("SalesOrderChangeFeed", leaseContainer);
+
+                using FeedIterator<ChangeFeedProcessorState> estimatorIterator = changeFeedEstimator.GetCurrentStateIterator();
                 int partitionCount = 0;
 
-                using (FeedIterator<ChangeFeedProcessorState> iterator = estimator.GetCurrentStateIterator())
+                long lagCount=0;
+                while (estimatorIterator.HasMoreResults)
                 {
-                    while (iterator.HasMoreResults)
+                    FeedResponse<ChangeFeedProcessorState> states = await estimatorIterator.ReadNextAsync();
+                   
+                    foreach (ChangeFeedProcessorState leaseState in states)
                     {
-                        FeedResponse<ChangeFeedProcessorState> states = await iterator.ReadNextAsync();
-                        partitionCount += states.Where(state => state.EstimatedLag > 0).Count();
+                        string host = leaseState.InstanceName == null ? $"not owned by any host currently" : $"owned by host {leaseState.InstanceName}";
+                        _logger.LogInformation($"Lease [{leaseState.LeaseToken}] {host} reports {leaseState.EstimatedLag} as estimated lag.");
+                        lagCount = lagCount + leaseState.EstimatedLag;
+
                     }
+                    partitionCount += states.Where(state => state.EstimatedLag > 0).Count();
                 }
 
+                s_CFRecordsReceived.Add(lagCount);
```

Comment:
I don't find `s_*` variables are used anywhere. Can you explain what do they achieve?